src/bun.js/test/snapshot.zig

```diff
@@ -0,0 +1,52 @@
+const bun = @import("bun");
+const strings = bun.strings;
+const fs = @import("../../fs.zig");
+const default_allocator = bun.default_allocator;
+
+
+const std = @import("std");
+const testing = std.testing;
+
+const SNAPSHOTS_DIR = "__snapshots__/";
+const EXTENSION = ".snap";
+
+pub fn resolveSnapshotPath(test_path: fs.Path) fs.Path {
+    const dir_path = test_path.sourceDir();
+    const file_name = test_path.name.base;
+    // A test's name could be `feature.test.ts` and the base is `feature.test`
+    // `test` should be stripped if it exists
+    const file_name_stripped = std.mem.trimRight(u8, file_name, ".test");
+
+    var string_list = &[_][]const u8{dir_path, SNAPSHOTS_DIR, file_name_stripped, EXTENSION };
+    const snapshot_file_path = strings.join(string_list, "", std.heap.page_allocator) catch unreachable;
```

Comment:
Instead of using a global allocator here, pass an `allocator: std.mem.Allocator` and in the test runner, use the allocator from `globalObject`